lib/swarm/registry.ex

```diff
@@ -11,6 +11,20 @@ defmodule Swarm.Registry do
   defdelegate register(name, pid), to: Tracker, as: :track
   defdelegate register(name, module, fun, args, timeout), to: Tracker, as: :track
 
+  @spec whereis_or_register_name(term, atom(), atom(), [term]) :: {:ok, pid} | {:error, term}
+  def whereis_or_register_name(name, m, f, a, timeout \\ :infinity)
+  @spec whereis_or_register_name(term, atom(), atom(), [term], non_neg_integer() | :infinity) :: {:ok, pid} | {:error, term}
+  def whereis_or_register_name(name, module, fun, args, timeout) do
+    case Swarm.whereis_name(name) do
+      :undefined ->
+        case Swarm.register_name(name, module, fun, args, timeout) do
+          {:ok, pid} -> pid
+          {:error, {:already_registered, pid}} -> pid
+        end
+      pid -> pid
+    end
```

Comment:
These case statements could be nested if you like.
```
case Swarm.whereis_name(name) do
  :undefined -> Swarm.register_name(name, module, fun, args, timeout)
  pid -> {:ok, pid}
end
|> case do
  {:error, {:already_registered, pid}} -> pid
  {:ok, pid} -> pid
  error -> error
end
```

This PR seems like a useful addition, I have almost identical code in my project.

Reply:
If anything, I would use `with` for flattening cases, after all that was why it was added to the language, but the nested cases are acceptable here as well since it's just one level of nesting:

```elixir
with :undefined <- whereis_name(name),
     {:ok, pid} <- register_name(name, module, fun, args, timeout) do
  {:ok, pid}
else
  pid when is_pid(pid) ->
    {:ok, pid}
  {:error, {:already_registered, pid}} ->
    {:ok, pid}
  {:error, _} = err ->
    err
end
```